File: android/app/src/main/kotlin/com/wrbug/gitbbs/flutterbridge/mmkv/method/SaveUserCall.kt
package com.wrbug.gitbbs.flutterbridge.mmkv.method

import com.wrbug.gitbbs.flutterbridge.FlutterMethodCall
import com.wrbug.gitbbs.mmkv.UserKv
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel


class SaveUserCall : FlutterMethodCall {
    override fun getMethodName() = "saveUser"

    override fun invoke(call: MethodCall, result: MethodChannel.Result) {
        if (call.hasArgument("user")) {
            val user = call.argument<String>("user") ?: ""
            UserKv.saveUser(user)
            result.success(null)
            return
        }
        result.error("miss user", "", "")
    }
}

File: ios/Runner/FlutterBridge/mmkv/method/SaveUserCall.swift
//
//  GetTokenCall.swift
//  Runner
//
//  Created by WrBug on 2019/3/7.
//  Copyright © 2019 The Chromium Authors. All rights reserved.
//

import Foundation
class SaveUserCall: MethodCall {
    func getMethodName() -> String {
        return "saveUser"
    }
    
     func invoke(call: FlutterMethodCall, result: (Any?) -> Void) {
        var args =  call.arguments as! Dictionary<String, Any>
        UserKv.saveUser(args["user"]as! String)
        result(true)
    }
}
